src/Transports/MassTransit.AmazonSqsTransport/Pipeline/AmazonSqsMessageReceiver.cs

```diff
@@ -46,7 +46,8 @@ public AmazonSqsMessageReceiver(ClientContext client, SqsReceiveEndpointContext
             _dispatcher = context.CreateReceivePipeDispatcher();
             _dispatcher.ZeroActivity += HandleDeliveryComplete;
 
-            Task.Run(Consume);
+            var task = Task.Run(Consume);
+            SetCompleted(task);

```

Comment:
it's not right, you shouldn't complete consume, until you wanna stop it

Reply:
That's not what SetCompleted does. If you look at the implementation, SetCompleted adds a continuation to the Consume loop task, so that once the Consume loop completes, the underlying completion logic of the agent can be carried on.

Reply:
this will prevent receiver transport to start. I suspect I've seen this issue when was working on kafka

Reply:
Just to be clear, this is working, all tests are passing -- the failed test in the build seems to be flakiness in the test case.  I'd be interested in understanding how this would prevent transport to start -- I fail to see how.

Reply:
it won't prevent, your transport will be unhealthy if you do it right.
There is general way how to handle exceptions in the transport and I believe SQS should follow them.

Reply:
But surely you want the transport to be unhealthy if it can't reach SQS, right? Perhaps there could be a local retry, but then the question is, how many times do you retry before failing the agent and becoming unhealthy?